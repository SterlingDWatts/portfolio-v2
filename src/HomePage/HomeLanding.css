.HomeLanding {
  height: 100vh;
  min-height: 533px;
  padding-top: 48px;
  display: flex;
  flex-direction: column;
  background-color: #00685b;
  color: white;
}

.HomeLanding__header {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 20%;
  min-height: 100px;
}

.HomeLanding h1 {
  font-size: 40px;
  line-height: 40px;
}

.HomeLanding__italy_image {
  height: 50%;
  min-height: 265px;
  background-image: url(https://sterlingdwatts.github.io/portfolio/images/italy.jpg);
  background-position: center;
  background-size: cover;
}

.HomeLanding__blurb {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 30%;
  min-height: 120px;
  padding: 0 16px;
}

@media screen and (min-width: 500px) {
  .HomeLanding__italy_image {
    width: 80%;
    margin: auto;
    box-shadow: 6px 6px 5px 0 rgba(0, 0, 0, 0.75);
  }

  .HomeLanding__blurb {
    width: 80%;
    margin: auto;
  }
}

@media screen and (min-width: 900px) {
  .HomeLanding__header {
    height: 60%;
    width: 50%;
  }

  .HomeLanding__italy_image {
    width: 50%;
    height: 100%;
    order: 90;
    box-shadow: none;
  }

  .HomeLanding__blurb {
    width: 50%;
    height: 40%;
    padding: 64px;
  }
}
