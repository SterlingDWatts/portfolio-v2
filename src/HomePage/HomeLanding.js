import React, { Component } from 'react';
import './HomeLanding.css';

class HomeLanding extends Component {

	render() {
		return (
			<section className='HomeLanding'>
				<div className='HomeLanding__header'>
					<header>
						<h1>Sterling Watts</h1>
						<h4>Full Stack Web Developer</h4>
					</header>
				</div>
				<div className='HomeLanding__italy_image'></div>
				<div className='HomeLanding__blurb'>
					<p>I am a passionate Full Stack Web Developer that enjoys making beautiful and intuitive websites as well as analyzing the data that I can get from them!</p>
				</div>
			</section>
		);
	}


}

export default HomeLanding;
