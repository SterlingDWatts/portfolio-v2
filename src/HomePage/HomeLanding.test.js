import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HomeLanding from './HomeLanding';

describe(`HomeLanding Component`, () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<HomeLanding />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer   
            .create(<HomeLanding />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

})
