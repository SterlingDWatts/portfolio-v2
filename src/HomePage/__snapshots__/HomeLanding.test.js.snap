// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeLanding Component renders the UI as expected 1`] = `
<section
  className="HomeLanding"
>
  <div
    className="HomeLanding__header"
  >
    <header>
      <h1>
        Sterling Watts
      </h1>
      <h4>
        Full Stack Web Developer
      </h4>
    </header>
  </div>
  <div
    className="HomeLanding__italy_image"
  />
  <div
    className="HomeLanding__blurb"
  >
    <p>
      I am a passionate Full Stack Web Developer that enjoys making beautiful and intuitive websites as well as analyzing the data that I can get from them!
    </p>
  </div>
</section>
`;
